### PR TITLE
Website file proof, handle protocal DESKTOP-1701

### PR DESCRIPTION
**Pull request by awendland**
## What this is

Currently, the UI expects a user to enter a file-based web proof as `alexwendland.com`, with only the hostname specified. However, the service can handle more elaborate URLs such as `http://alexwendland.com` or even `ftp://alexwendland.com/test`. Both of these latter cases are actually interpretted as HTTPS proofs for `alexwendland.com`, because the service extracts the hostname from the provided URL and then tests ports 80 and 443 for a response. The service will accept this endpoints and convert them to the hostname, but the GUI isn't aware of this conversion and therefore still tries to use `ftp://alexwendland.com/test` as the hostname, which results in recommending that the user host their txt file proof at `http://fpt/keybase.txt` which makes no sense. This PR changes the `updateUsername` reducer to attempt to extract a hostname from the provided input, if it's HTTPS or HTTP, to avoid these problems.

I've added a new action for processing the username once the user has clicked the continue button. This allows the current method of storing every character change in the store to stay.

*Note: nothing is done to DNS username text because the service only allows hostnames to be provided so it will error on the aforementioned cases that cause issues with HTTP(S).*

## Some PR nastyness

To get this out for review quickly, I've opened it against another branch because testing it is dependent on those changes. Once/if this gets thumbed I will reopen it against master and merge it in directly once CI passes.

@keybase/react-hackers

### CODE REVIEW

File: shared/profile/prove-enter-username.desktop.js [hunk 1]
```diff
@@ -81,9 +81,6 @@ class Render extends Component<void, Props, State> {
   }
 
   handleUsernameChange (username: string) {
-    if (this.props.onUsernameChange) {
```
Comment (chrisnojima) [inline shared/profile/prove-enter-username.desktop.js]: Why was this removed

Comment (chrisnojima) [inline shared/profile/prove-enter-username.desktop.js]: Doesn't this break the Bitcoin address checking

Comment (awendland) [inline shared/profile/prove-enter-username.desktop.js]: One sec, double checking

Comment (awendland) [inline shared/profile/prove-enter-username.desktop.js]: Yes, it does break BTC. I'll update with a fix for that.

It was removed from triggering every time the user typed to only when the user clicked submit. Reason being so that the hostname extraction stuff wasn't running every time and storing the extracted hostname state in a different part of the store. I could add a separate step to the submitUsername action that does some hostname cleanup prior to sending it to the service/storing it for the next screen, or we could move the BTC validation and other validation stuff into the components internal state.
